python/ray/actor.py

```diff
@@ -129,34 +127,112 @@ def annotate_method(method):
     return annotate_method
 
 
+class ActorMethodShell:
+    """A class used to invoke an actor method.
+
+    This class holds static metadata about the actor method, such as
+    the method name, number of return values, and whether the method is a
+    generator. It does not hold a reference to the actor handle.
+    This is used to create ActorMethod objects that hold a strong reference
+    to the actor handle for the duration of the invocation.
+    """
```

Comment:
```suggestion
    """A container for the metadata required to invoke an actor method.
    
    This class intentionally does *not* hold a reference to the `ActorHandle`, as that causes
    a circular reference that delays `ActorHandle` destruction until the Python GC runs.
    
    Instead, it can be used as a factory to lazily generate `ActorMethod` instances that can be used to submit actor tasks for this method.
    """
```